ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -1024,6 +1052,8 @@ private void endCompaction() {
       copyCandidates.removeIf(
           copyCandidate -> isDuplicate(copyCandidate, duplicateSearchSpan, indexSegment.getStartOffset(),
               checkAlreadyCopied));
+      copyCandidates.removeIf(
+          copyCandidate -> isDeleteRequested(copyCandidate, selectedContainers));
```

Comment:
Can we combine this with line 1052-1053?

Reply:
Updated.